sdk/resourcemanager/Azure.ResourceManager.Core/src/ArmException.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Runtime.Serialization;
+using Azure.Core;
+
+#nullable enable
+
+namespace Azure.ResourceManager.Core
+{
+    /// <summary>
+    /// A class representing exceptions thrown by Azure ARM services.
+    /// </summary>
+    [Serializable]
+    public class ArmException : RequestFailedException
+    {
+        /// <summary>
+        /// Gets and set the Code of the ARM exception.
+        /// </summary>
+        public string Code { get; internal set; }
```

Comment:
RequestFailedException already has an `ErrorCode` property having both might be confusing.

Reply:
Arm Error (Odata error) has Code, Message etc. It is direct mapping. I would have preferred the RequestFailed to be HttpStatusCode, but...